docs/examples/classification.ipynb

```diff
@@ -0,0 +1,946 @@
+   "source": [
+    "%%time\n",
+    "# Note that this cell might take a little while.\n",
+    "from sklearn.preprocessing import OrdinalEncoder\n",
+    "from sklearn.experimental import enable_halving_search_cv\n",
+    "from sklearn.model_selection import HalvingGridSearchCV\n",
+    "from sklearn.ensemble import HistGradientBoostingClassifier\n",
+    "\n",
+    "\n",
+    "\n",
+    "col_trans_bt = ColumnTransformer(\n",
+    "    [\n",
+    "        (\"categorical_features\", OrdinalEncoder(), x_discrete),\n",
+    "    ]\n",
+    ")\n",
+    "m_hgbt_base = Pipeline(\n",
+    "    [\n",
+    "        (\"column_transformer\", col_trans_bt),\n",
+    "        (\n",
+    "            \"model\", HistGradientBoostingClassifier(\n",
+    "                loss=\"log_loss\",\n",
+    "                categorical_features=list(range(len(x_discrete))),\n",
+    "                max_iter=200,\n",
+    "                random_state=33,\n",
+    "            )\n",
+    "        ),\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "param_grid = {\n",
+    "    \"model__learning_rate\": [0.02, 0.05],\n",
+    "    \"model__l2_regularization\": [0.1, 0.2, 0.3, 1],\n",
+    "    \"model__max_depth\": [None, 1, 2, 3, 5],\n",
+    "    \"model__max_leaf_nodes\": [3, 10, 30, 50]\n",
+    "}\n",
+    "\n",
+    "\n",
+    "# successive halfing grid search (CV) on the training data\n",
+    "shgs = HalvingGridSearchCV(\n",
+    "    m_hgbt_base,\n",
+    "    param_grid=param_grid,\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_log_loss\",\n",
+    "    random_state=321,\n",
+    ").fit(X_train, y_train)\n",
+    "\n",
+    "m_hgbt = shgs.best_estimator_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "({'model__l2_regularization': 0.1,\n",
+       "  'model__learning_rate': 0.05,\n",
+       "  'model__max_depth': None,\n",
+       "  'model__max_leaf_nodes': 50},\n",
+       " -0.012953509513355315)"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "shgs.best_params_, shgs.best_score_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "200"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "m_hgbt[-1].n_iter_"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Calibration Assessment\n",
+    "To make the code easier, we put together the predictions of all of our models on the test set in a single dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _helper(mod, X): return pd.Series(mod.predict_proba(X)[:, 1])\n",
```

Comment:
```python
def predict_proba_helper(mod, X):
    "Only return the probability of the positive class."""
    return mod.predict_proba(X)[:, 1]
```
I think it is not necessary to convert to pd.Series here.

Reply:
Currently the models are returning numpy arrays. I tried to add .set_output(transform="pandas") to the pipelines without success. I would copy your function and add pd.Series in the return statement.

Reply:
It is totally fine for this function to retun a numpy array. Please don't convert to pd.Series.
Could we please also get rid of the clipping?
Maybe, I have to fix the logloss to return np.inf instead of np.nan.

Reply:
Done, it is now so:

```python3
models = [m_trivial, m_logistic, m_tree, m_hgbt]
df_pred_test = pd.DataFrame(
    np.array([m.predict_proba(X_test)[:, 1] for m in models]).T, columns=["Trivial", "Logistic", "Tree", "HGBT"]
    )
```